inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -703,3 +703,39 @@ def increase_restart_count_or_error(obj, eng):
         save_workflow(obj, eng)
     else:
         raise WorkflowsError('Workflow restarted too many times')
+
+
+def flatten_list(input_list):
+    if isinstance(input_list, (list, tuple)):
+        return [
+            element for innerList in input_list for element in flatten_list(innerList)
+        ]
+    return [input_list]
+
+
+@with_debug_logging
+def affiliations_for_hidden_collections(obj):
+    affiliations_mapping = current_app.config.get("AFFILIATIONS_TO_HIDDEN_COLLECTIONS_MAPPING", {})
+    affiliations = flatten_list(get_value(obj.data, 'authors.raw_affiliations.value'))
+
+    affiliations_set = {}
+    for aff in affiliations:
+        affiliations_set.update(aff.split())
+    aff_keys = set(affiliations_mapping.keys())
+    spotted_affiliations = aff_keys.intersection(affiliations_set)
+    return [affiliations_mapping[affiliation] for affiliation in spotted_affiliations]
```

Comment:
shouldn't be the check case insensitive?

Reply:
I have no idea actually. There was nothing about this in the task actually. @michamos  ?

Reply:
yes, better make the lookup case insensitive.